Makefile

```diff
@@ -25,9 +21,9 @@ ghc:
 # ------------------------------------------------------
 
 ## Builds hie for GHC version % only
-hie-%: submodules cabal
+hie-%: STACKLOCALBINDIR=$(shell stack --stack-yaml=stack-$*.yaml path --local-bin)
+hie-%: submodules cabal-%
 	stack --stack-yaml=stack-$*.yaml install happy
-	stack --stack-yaml=stack-$*.yaml build
```

Comment:
We do the build before install because it should be a no-op, but all 'stack install' checks is that the artifact exists, so will not rebuild changes if this is an updated workspace.